services/api-server/src/simcore_service_api_server/utils/client_base_dev.py

```diff
@@ -0,0 +1,58 @@
+import json
+import logging
+from pathlib import Path
+
+import httpx
+from fastapi.encoders import jsonable_encoder
+from httpx._types import URLTypes
+from jsonschema import ValidationError
+from pydantic import parse_file_as
+from simcore_service_api_server.utils.http_calls_capture import HttpApiCallCaptureModel
+
+from .http_calls_capture import get_captured
+from .http_calls_capture_processing import CaptureProcessingException
+
+_logger = logging.getLogger(__name__)
+
+
+class AsyncClientForDevelopmentOnly(httpx.AsyncClient):
+    """
+    Adds captures mechanism
+    """
+
+    def __init__(self, capture_file: Path, **async_clint_kwargs):
+        super().__init__(**async_clint_kwargs)
+        assert capture_file.name.endswith(
```

Comment:
This is only run in dev model (with optimization)

Reply:
this guy is a bot, it did not undertsand you ;-)
the problem is that codeclimate cannot understand multiline, add  `#nosec` at the end of the same line as `assert`
```python 

assert capture_file.name.endswith( # nosec
...
)
```

PS: we have to deactivate codeclimate since sonarcloud is already covering all this

Reply:
🤣 I guessed `codeclimate` is a bot. But I thought it would be good to answer in case someone (meaning: a human) comes back and looks at this one day.

Reply:
I have added the nosec